File: web/forjix-web/src/app/core/api/inventory.models.ts
export type InventoryStatus = 'Normal' | 'Low' | 'OutOfStock' | 'Negative';
export type InventoryMovementType = 'StockEntry' | 'StockExit' | 'PositiveAdjustment' | 'NegativeAdjustment';
export interface InventoryItem { productId: string; productName: string; sku: string; barcode?: string; categoryId: string; categoryName: string; quantity: number; minimumStock: number; costPrice: number; salePrice: number; status: InventoryStatus; lastMovementAt?: string; rowVersion: string; }
export interface InventoryMovementItem { id: number; productId: string; type: InventoryMovementType; quantity: number; previousQuantity: number; newQuantity: number; reason?: string; referenceType?: string; referenceId?: string; userId: string; userName: string; createdAt: string; }
export interface CreateInventoryMovementRequest { type: InventoryMovementType; quantity: number; reason?: string | null; rowVersion?: string; }
export interface InventoryMovementResult { inventory: InventoryItem; movement: InventoryMovementItem; }
export interface PagedResult<T> { items: T[]; page: number; pageSize: number; total: number; }

File: src/Forjix.Application/Features/Inventory/InventoryModels.cs
namespace Forjix.Application.Features.Inventory;

public sealed record InventoryItem(
    Guid ProductId, string ProductName, string Sku, string? Barcode, Guid CategoryId,
    string CategoryName, decimal Quantity, decimal MinimumStock, decimal CostPrice, decimal SalePrice,
    string Status, DateTimeOffset? LastMovementAt, string RowVersion);

public sealed record InventoryMovementItem(
    long Id, Guid ProductId, string Type, decimal Quantity, decimal PreviousQuantity,
    decimal NewQuantity, string? Reason, string? ReferenceType, string? ReferenceId,
    Guid UserId, string UserName, DateTimeOffset CreatedAt);

public sealed record CreateInventoryMovementRequest(string Type, decimal Quantity, string? Reason, string? RowVersion);
public sealed record InventoryMovementResult(InventoryItem Inventory, InventoryMovementItem Movement);
public sealed record PagedResult<T>(IReadOnlyList<T> Items, int Page, int PageSize, int Total);
